features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/filters/RoomListFiltersView.kt

```diff
@@ -0,0 +1,173 @@
+
+@Composable
+fun RoomListFiltersView(
+    state: RoomListFiltersState,
+    modifier: Modifier = Modifier
+) {
+    fun onClearFiltersClicked() {
+        state.eventSink(RoomListFiltersEvents.ClearSelectedFilters)
+    }
+
+    fun onFilterClicked(filter: RoomListFilter) {
+        state.eventSink(RoomListFiltersEvents.ToggleFilter(filter))
+    }
+
+    val startPadding = if (state.hasAnyFilterSelected) 4.dp else 16.dp
+    Row(
+        modifier = modifier.padding(start = startPadding, end = 16.dp),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        AnimatedVisibility(visible = state.hasAnyFilterSelected) {
+            RoomListClearFiltersButton(
+                modifier = Modifier.testTag(TestTags.homeScreenClearFilters),
+                onClick = ::onClearFiltersClicked
+            )
+        }
+        val lazyListState = rememberLazyListState()
+        val fadingEdgesBrush = horizontalFadingEdgesBrush(
+            showLeft = lazyListState.canScrollBackward,
+            showRight = lazyListState.canScrollForward
+        )
+        LazyRow(
+            modifier = Modifier
+                .fillMaxWidth()
+                .fadingEdge(fadingEdgesBrush),
+            state = lazyListState,
+            horizontalArrangement = Arrangement.spacedBy(8.dp)
+        ) {
+            roomListFilters(state.selectedFilters, selected = true, onClick = ::onFilterClicked)
+            roomListFilters(state.unselectedFilters, selected = false, onClick = ::onFilterClicked)
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+private fun LazyListScope.roomListFilters(
+    filters: ImmutableList<RoomListFilter>,
+    selected: Boolean,
+    onClick: (RoomListFilter) -> Unit,
+) {
+    items(
+        items = filters,
+        key = { it.ordinal },
+    ) { filter ->
+        RoomListFilterView(
+            modifier = Modifier.animateItemPlacement(),
+            roomListFilter = filter,
+            selected = selected,
+            onClick = onClick,
+        )
+    }
+}
+
+@Composable
+private fun RoomListClearFiltersButton(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    IconButton(
+        modifier = modifier,
+        onClick = onClick,
+    ) {
+        Box(
+            modifier = Modifier
+                .clip(CircleShape)
+                .background(ElementTheme.colors.bgActionPrimaryRest)
+        ) {
+            Icon(
+                modifier = Modifier.align(Alignment.Center),
+                imageVector = CompoundIcons.Close(),
+                tint = ElementTheme.colors.iconOnSolidPrimary,
+                contentDescription = stringResource(id = io.element.android.libraries.ui.strings.R.string.action_clear),
+            )
+        }
+    }
+}
+
+@Composable
+private fun RoomListFilterView(
+    roomListFilter: RoomListFilter,
+    selected: Boolean,
+    onClick: (RoomListFilter) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    FilterChip(
```

Comment:
Do we need to add this component in the Compound library at some point?

Reply:
Yes it'll probably be, but for now we can stay simple.